scio-coders/src/main/scala/com/spotify/scio/coders/Coder.scala

```diff
@@ -45,15 +47,23 @@ sealed trait Coder[T] extends Serializable
 final case class Beam[T] private (beam: BCoder[T]) extends Coder[T]
 final case class Fallback[T] private (ct: ClassTag[T]) extends Coder[T]
 final case class Transform[A, B] private (c: Coder[A], f: BCoder[A] => Coder[B]) extends Coder[B]
-final case class Disjunction[T, Id] private (idCoder: Coder[Id],
+final case class Disjunction[T, Id] private (typeName: String,
+                                             idCoder: Coder[Id],
                                              id: T => Id,
                                              coder: Map[Id, Coder[T]])
     extends Coder[T]
-final case class Record[T] private (cs: Array[(String, Coder[T])]) extends Coder[Array[T]]
+
+final case class Record[T] private (typeName: String,
```

Comment:
`typeName: String` doesn't look good, but don't know of a better serializable type, don't want to add dependency to `magnolia.TypeName`